src/sync/pull.rs

```diff
@@ -1343,4 +1459,175 @@ mod tests {
             };
         }
     }
+
+    #[async_std::test]
+    async fn test_changed_keys() {
```

Comment:
@phritz I cannot figure out how to write this test.

I want to have the snapshot contain a non empty map and an index. Then I want to run begin and end pull with a patch that would update the index. Then I want to check the diff.

Maybe the test is fine and the implementation is wrong? I just can't figure this out and I have spent way too much time on this test already.

Please help